orttraining/orttraining/test/python/orttraining_test_orttrainer_checkpoint_functions.py

```diff
@@ -0,0 +1,351 @@
+import pytest
+from unittest.mock import patch, Mock
+from orttraining_test_orttrainer_frontend import _load_pytorch_transformer_model
+from onnxruntime.training import amp, checkpoint, optim, orttrainer
+import numpy as np
+import onnx
+import torch
+
+# Helper functions
+
+def _create_trainer(zero_enabled=False):
+    """Cerates a simple ORTTrainer for ORTTrainer functional tests"""
+
+    device = 'cuda'
+    optim_config = optim.LambConfig(lr=0.1)
+    opts = {
+                'device' : {'id' : device},
+                'debug' : {'deterministic_compute': True}
+            }
+    if zero_enabled:
+        opts['distributed'] = {
+                'world_rank' : 0,
+                'world_size' : 1,
+                'allreduce_post_accumulation' : True,
+                'deepspeed_zero_optimization':
+                {
+                    'stage': 1
+                }
+            }
+    model, model_desc, loss_fn, batcher_fn, train_data, _, _ = _load_pytorch_transformer_model(device)
+    trainer = orttrainer.ORTTrainer(model, model_desc, optim_config, loss_fn=loss_fn, options=orttrainer.ORTTrainerOptions(opts))
+
+    return trainer
+
+class _training_session_mock(object):
+    """Mock object for the ORTTrainer _training_session member"""
+
+    def __init__(self, model_states, optimizer_states, partition_info):
+        self.model_states = model_states
+        self.optimizer_states = optimizer_states
+        self.partition_info = partition_info
+
+    def get_model_state(self, include_mixed_precision_weights=False):
+        return self.model_states
+
+    def get_optimizer_state(self):
+        return self.optimizer_states
+
+    def get_partition_info_map(self):
+        return self.partition_info
+
+def _get_load_state_dict_strict_error_arguments():
+    """Build parameterized list of arguments to test strict loading of the state dictionary"""
```

Comment:
docstring didnt help much. why is it required? how it woks?
Review private dosctring so that they explain the idea behind the code and why they are needed
Review public docstring to make sure it doesnt bring implementation details to the users, just an overall input this and I return that

Reply:
Basically, this helper function creates the input arguments to test the ```strict``` loading of state dictionaries. That test is implemented as a parameterized test, so it needs a list of parameters to test. This helper function creates that list and returns it for that test.
I will add more details to the docstring.